### PR TITLE
feat: initialize storage

### PR DESCRIPTION
**Pull request by Freyskeyd**
# Description

Initialize the storage crate to persist some state of the agglayer.

## PR Checklist:

- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added or updated tests that comprehensively prove my change is effective or that my feature works

### CODE REVIEW

File: crates/agglayer-storage/src/columns/latest_certificate_per_network/mod.rs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+use super::{
+    default_bincode_options, Codec, ColumnSchema, LATEST_SETTLED_CERTIFICATE_PER_NETWORK_CF,
+};
+use crate::{error::Error, types::NetworkId};
+
+#[cfg(test)]
+mod tests;
+
+/// Column family for the lastest settled certificate per network.
```
Comment (hadjiszs) [inline crates/agglayer-storage/src/columns/latest_certificate_per_network/mod.rs:12]:
```suggestion
/// Column family for the latest settled certificate per network.
```

File: crates/agglayer-storage/src/stores/state/mod.rs [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    }
+}
+
+impl StateReader for StateStore {
+    /// Get the active networks.
+    /// Meaning, the networks that have at least one submitted certificate.
+    ///
+    /// Performance: O(n) where n is the number of networks.
+    /// This is because we need to scan all the keys in the
+    /// `last_certificate_per_account` column family.
```
Comment (hadjiszs) [inline crates/agglayer-storage/src/stores/state/mod.rs:29]: I don't find such column, do you mean `latest_settled_certificate_per_network`?

I think the related benchmark `last_certificate_bench.rs` also refer to a different name of this column

**Review by hadjiszs — APPROVED** [review]

File: crates/agglayer-storage/src/types.rs [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+impl Deref for Height {
+    type Target = u64;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+#[derive(Debug, PartialEq, Eq, Serialize, Deserialize)]
+pub struct NetworkId(pub(crate) u32);
```
[the remark this answers is not in the document]
Comment (Freyskeyd) [reply crates/agglayer-storage/src/types.rs:88]: Yes, the types are moved to a future `agglayer-types` in a next PR

**Comment by Freyskeyd** [comment]
this pr is superseded by multiple other PRs

**Pull request closed without merging**